core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3053,6 +3053,22 @@ public static double acos(double b0) {
     return Math.acos(b0);
   }
 
+  // ACOSD
+  /** SQL <code>ACOSD</code> operator applied to BigDecimal values. */
+  public static double acosd(BigDecimal b0) {
+    return acosd(b0.doubleValue());
+  }
+
+  /** SQL <code>ACOSD</code> operator applied to double values. */
+  public static double acosd(double b0) {
+    final double radians = Math.acos(b0);
+    if (Double.isFinite(radians)) {
```

Comment:
How about extracting all these blocks of:
```
if (Double.isFinite(radians)) {
  return Math.toDegrees(radians);
} else {
  return Double.NaN;
}
```
into a single common auxiliary method?

Reply:
The methods have been reworked and this doesn't apply. There is perhaps room for a similar change for "cosd", "sind" and "tand", but they are rather small.